### ISSUE TITLE
Autocomplete on macOS interferes with typing

### ISSUE THREAD

**Issue by str4d** [Electric-Coin-Company/zashi-ios#1013]
### Describe the issue
The macOS keyboard's autocomplete is interfering with the ability to correctly type content into the app.

#### If so, please list the steps to reproduce below:
1. Install Zashi iOS on an ARM64 Macbook.
2. Go to the send screen.
3. Start typing into the memo field.

### Expected behaviour
What I type should be what ends up in the memo field.

### Actual behaviour + errors
Every so often parts of words disappear. When typing slowly, it appears that macOS attempts to autocomplete a word, and then the next letter typed replaces the first part of the word instead of being appended to it.

- App Version: 0.2.0 (14)
- iOS Version: macOS 14.2.1
- Device: (if applies) Macbook Air M1.

**Comment by zerodartz** [comment]
Maybe not related, but I have very similar, maybe same bug on my iPhone every so often on not Zashi but random apps using both default keyboard or some others. So just maybe it's a bigger issue somewhere deeper not Zashi?

**Comment by LukasKorba** [comment]
> Maybe not related, but I have very similar, maybe same bug on my iPhone every so often on not Zashi but random apps using both default keyboard or some others. So just maybe it's a bigger issue somewhere deeper not Zashi?

I think this is a system level bug, nothing that can be fixed on Zashi level. But will investigate it at some point in near future. @true-jared is this a Zashi 1.0 hard requirement? If you read the desc, it's a case when Zashi is installed on a macbook, I don't think this is considered a valid case for Zashi 1.0.

**Comment by true-jared** [comment]
I agree @LukasKorba, if this is not an issue/blocker on the mobile device, we should push this to post-Zashi 1.0 and investigate it later on.

**Comment by daira** [comment]
A user can probably work around this by [turning off "inline predictive text"](https://www.macrumors.com/how-to/how-to-disable-macos-inline-predictive-text/) globally. That said, it's obviously still a bug in the app, relative to other macOS apps.

Another possibility is to [turn off autocomplete in the app](https://stackoverflow.com/a/70005475/393146) for all text fields that it affects:
```swift
textField.autocorrectionType = .no
textField.spellCheckingType = .no
```

**Comment by LukasKorba** [comment]
Fixed in Zashi 1.2
